### ISSUE TITLE
GestureRecognizers are too "sloppy" for accurate pointing devices

### ISSUE THREAD

**Issue by shyndman** [flutter/flutter#36381]
The slop constants in [`src/gestures/constants.dart`](https://github.com/flutter/flutter/blob/2e1a8c746f4d170ceff362315f9933d0aa50d1b7/packages/flutter/lib/src/gestures/constants.dart#L61) are used to determine the activation threshold for various `GestureRecognizer`s.

They're great for touch input, but feel inappropriately "sloppy" when using the mouse/trackpad.

Testing locally, I've been able to lower the slops considerably without mis-activating gestures.

**Comment by gspencergoog** [comment]
Which recognizers are you talking about here?

It does seem like we should have different values for different input mechanisms, (e.g. touchpad vs mouse).  We don't currently support touchpad gestures, so when we add that we'll need to differentiate anyhow.

**Comment by shyndman** [comment]
Pan and the drags are particularly noticeable.

**Comment by maxim-smotrov** [comment]
To add some clarification: the threshold for drag is defined as
`const double kPanSlop = kTouchSlop * 2.0;`
which calculates to 36. So the GestureDetector.onPanStart fires only when the cursor travels more than 36 pixels which is a huge distance for a mouse pointer.
For example, when I try to select some text it skips 2-3 characters before starting the actual selection.

**Comment by orestesgaolin** [comment]
This is one of the small things that make bad impression when running Flutter on desktop. Is there any way to override this behavior?

**Comment by markusaksli-nc** [comment]
Reproducible on the latest master `1.21.0-6.0.pre.226`.
<details>
 <summary>Minimal reproducible code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Sample',
      home: Scaffold(
        appBar: AppBar(
          title: Text('Sample'),
        ),
        body: Container(
          child: Center(
            child: TextField(),
          ),
        ),
      ),
    );
  }
}
```
</details>

Pointer can be dragged over 3 letters before selection starts.
<img src="https://i.gyazo.com/89f4bba4402ce4eef4f2d42eac5768d9.gif" height="400" />
<details>
 <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.21.0-6.0.pre.226, on Mac OS X 10.15.6 19G73, locale en-GB)
    • Flutter version 1.21.0-6.0.pre.226 at /Users/markus/development/flutter_master
    • Framework revision fa60f525f0 (2 hours ago), 2020-08-06 08:01:03 -0400
    • Engine revision 36db6cfdd1
    • Dart version 2.10.0 (build 2.10.0-3.0.dev 14a6aac97b)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at /Users/markus/Library/Android/sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: /Users/markus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Users/markus/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/193.6626763/Android Studio.app/Contents
    • Flutter plugin version 48.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (5 available)
    • sdk gphone x86 arm (mobile) • emulator-5554                            • android-x86    • Android 11 (API 30) (emulator)
    • Nevercode’s iPhone (mobile) • b668e524315069f3db3661ac11ff1f66afafebdb • ios            • iOS 13.6
    • macOS (desktop)             • macos                                    • darwin-x64     • Mac OS X 10.15.6 19G73
    • Web Server (web)            • web-server                               • web-javascript • Flutter Tools
    • Chrome (web)                • chrome                                   • web-javascript • Google Chrome 84.0.4147.105

• No issues found!
```
</details>

**Comment by csells** [comment]
Here's another case of the same thing:

## Steps to Reproduce
1. Download the flutter-go sample
2. Run
3. Drag a piece onto the board
4. Drag a piece around the board w/ the mouse

**Expected results:**
The drag should happen immediately on a click 'n' drag.

**Actual results:**
The drag doesn't start until the mouse is moved 5-10 pixels:

![sloppy_mouse_drag_n_drop](https://user-images.githubusercontent.com/2568253/90202069-70c24e00-dd91-11ea-8d2c-7896628509a6.gif)

**Comment by csells** [comment]
cc @goderbauer

**Comment by goderbauer** [comment]
We need separate slops for high-accuracy devices and should not be using the touchSlope constants for those.

cc @dkwingsmt

**Comment by jonahwilliams** [comment]
This might be partially solved by https://github.com/flutter/flutter/pull/64267 , although as noted in the PR description I did not change scroll physics, which will need to be handled separately

**Comment by goderbauer** [comment]
I believe this particular issue is fixed by #64267. Closing.
